pkg/apis/operator/v1alpha1/tektonconfig_types.go

```diff
@@ -24,11 +24,6 @@ import (
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
-var (
-	_ TektonComponent     = (*TektonConfig)(nil)
-	_ TektonComponentSpec = (*TektonConfigSpec)(nil)
-)
-
```

Comment:
what makes TektonConfig not implement the TektonComponent Interface.
If TetkonConfig does not implement TektonComponent Interface, will that restrict us in some way in tests or in extensions.

Reply:
we can enable it back but with some changes I have added a TODO in lifecycle for it
once we refactor al components (dashboard, result ) we can find common things and update TektonComponent
otherwise they would break

Reply:
could you add it as an issue in https://github.com/tektoncd/operator/issues

Reply:
my concern is that, if we don't have this checks, we might end up loosing uniformity in methods of components in the future. It is not a deal breaker, but i would like a reminder to check this later.

Reply:
https://github.com/tektoncd/operator/issues/424
created an issue